xcube_zenodo/store.py

```diff
@@ -142,14 +161,86 @@ def get_open_data_params_schema(
     def open_data(
         self, data_id: str, opener_id: str = None, **open_params
     ) -> xr.Dataset:
-        uri = translate_data_id2uri(data_id)
-        return self._https_data_store.open_data(
-            data_id=uri, opener_id=opener_id, **open_params
+        format_id = estimate_file_format(data_id)
+        if format_id in COMPRESSED_FORMATS:
+            if not self.cache_store.has_data(data_id):
+                raise DataStoreError(
+                    f"The dataset {data_id} is stored in a compressed format. "
+                    f"Please use store.preload_data({data_id}) first."
+                )
+            return self.cache_store.open_data(
+                data_id=data_id, opener_id="dataset:zarr:file", **open_params
+            )
+        elif self.cache_store.has_data(data_id):
+            return self.cache_store.open_data(data_id=data_id, **open_params)
+        else:
+            uri = translate_data_id2fs_path(data_id)
+            return self._https_data_store.open_data(
+                data_id=uri, opener_id=opener_id, **open_params
+            )
+
+    def preload_data(self, *data_ids: str, **preload_params) -> PreloadHandle:
+        schema = self.get_preload_data_params()
+        schema.validate_instance(preload_params)
+        preload_params["merge"] = preload_params.get("merge", False)
+        preload_params["monitor_preload"] = preload_params.get("monitor_preload", True)
+        data_ids_sel = []
+        for data_id in data_ids:
+            format_id = estimate_file_format(data_id)
+            data_id_mod = data_id.replace(f".{format_id}", "/")
+            list_data_ids = self.cache_store.list_data_ids()
+            list_data_ids_mod = [i for i in list_data_ids if data_id_mod in i]
+            if list_data_ids_mod:
+                LOG.info(
+                    f"{data_id} is already pre-loaded. The datasets can be "
+                    f"opened with the following data IDs: "
+                    f"\n{'\n'.join(str(item) for item in list_data_ids_mod)}"
+                )
+            elif self.cache_store.has_data(data_id):
+                LOG.info(f"{data_id} is already pre-loaded.")
```

Comment:
What is the difference between the `if` and `elif` condition here? They both tell the user that the data_id is already pre-loaded.

Reply:
It depends on the compressed file. If only one dataset is in the compressed file, or all datasets are merged (this is a preload parameter), then the data_id stays. If multiple files are in the compressed file which cannot be merged, then the data_id is extended by the file name. Therefore, two cases in the `if` and `elif`.

Reply:
In the first case, I want to tell the user the extended data IDs.